### ISSUE TITLE
Error with deploying on Heroku:  KeyError: 'smtp_host is not a valid configuration key'

### ISSUE THREAD

**Issue by mmosleh** [Dallinger/Dallinger#1321]
I am trying to run the experiment on sanbox using dallinger sandbox --verbose however the deployment crashes. Looking at logs on Heroku using heroku logs --app APP returns the following error:

```
2018-08-26T23:03:38.022386+00:00 app[web.1]: /app/.heroku/python/lib/python3.6/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
2018-08-26T23:03:38.022410+00:00 app[web.1]: """)
2018-08-26T23:03:38.896183+00:00 app[web.1]: Traceback (most recent call last):
2018-08-26T23:03:38.896472+00:00 app[web.1]: File "launch.py", line 9, in <module>
2018-08-26T23:03:38.896691+00:00 app[web.1]: launch()
2018-08-26T23:03:38.896748+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/dallinger/experiment_server/gunicorn.py", line 85, in launch
2018-08-26T23:03:38.896987+00:00 app[web.1]: config.load()
2018-08-26T23:03:38.897018+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/dallinger/config.py", line 237, in load
2018-08-26T23:03:38.897334+00:00 app[web.1]: self.load_from_file(localConfig)
2018-08-26T23:03:38.897364+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/dallinger/config.py", line 198, in load_from_file
2018-08-26T23:03:38.897654+00:00 app[web.1]: self.extend(data, cast_types=True, strict=True)
2018-08-26T23:03:38.897907+00:00 app[web.1]: raise KeyError('{} is not a valid configuration key'.format(key))
2018-08-26T23:03:38.897684+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/dallinger/config.py", line 111, in extend
2018-08-26T23:03:38.897948+00:00 app[web.1]: KeyError: 'smtp_host is not a valid configuration key'
2018-08-26T23:03:31.010214+00:00 app[api]: Scaled to console@0:Hobby web@1:Hobby worker@1:Hobby by user 
2018-08-26T23:03:39.129347+00:00 heroku[web.1]: State changed from starting to crashed
```

## Current Behavior
Received the error described above

## Steps to Reproduce (for bugs)
1. create the dallingerconfig using dallinger setup and filled out aws access keys ,... : 
```
[AWS Access]

aws_access_key_id = xxxxxx
aws_secret_access_key = xxxxxx
aws_region = us-east-1

[Email Access]
smtp_host = smtp.gmail.com:587
smtp_username = xxxxxx
smtp_password =xxxxxx
```

2. use dyno type=hobby and heroku team on the config.txt 
3. run dallinger sanbox --verbose for bartlett1932 demo
4.

## Your Environment
* Dallinger version used:
4.0.0
* Operating System and version (desktop or mobile):
desktop
* Browser and version (e.g. Chrome 39, node.js 5.4):
Chrome
* Link to your project:

**Comment by MatthewWilkes** [comment]
Hi @mmosleh,

It appears that you're using an unreleased version of Dallinger locally. The `smtp_host`, `smtp_username` and `smtp_password` configuration variables were added after version 4.0.0 was released. Are you doing development work on Dallinger itself? If not, I'd recommend you install version 4.0.0 on your machine (https://dallinger.readthedocs.io/en/latest/installing_dallinger_for_users.html might be helpful).

In the meantime, there are two potential workarounds:

1) If you aren't relying on bug fixes since 4.0.0 that haven't been released yet, you can use the older configuration variables. They were changed to allow use of SMTP servers other than `smtp.gmail.com:587`, so your values will still work.

You should set:

    dallinger_email_address = xxx
    dallinger_email_key = xxx

instead of `smtp_username` and `smtp_password`. Omit the host line entirely. Also, you should be aware that your password will have to be an app password if you have 2 factor auth enabled.

2) If you do need the current unreleased version, you can change the requirements.txt file of the experiment you're running to:

    -e git://github.com/Dallinger/Dallinger.git#egg=dallinger

which will force dallinger to be installed from github rather than a released version. I'd recommend you avoid this unless you absolutely need it, as there might be more incompatible changes added later.

**Comment by mmosleh** [comment]
@MatthewWilkes  Awesome! thank you for your reply. It solved the problem.
